# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ron/tests/256_comma_error.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/ron/tests/256_comma_error.rs
--- a/third_party/rust/ron/tests/256_comma_error.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/ron/tests/256_comma_error.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -1,6 +1,9 @@
 #![allow(dead_code)]
 
-use ron::error::{Error, Position, SpannedError};
+use ron::error::{Error, Position, Span, SpannedError};
+
+#[cfg(feature = "internal-span-substring-test")]
+use ron::util::span_substring::check_error_span_inclusive;
 
 #[derive(Debug, serde::Deserialize)]
 struct Test {
@@ -19,7 +22,10 @@
         ron::from_str::<(i32, i32)>(tuple_string).unwrap_err(),
         SpannedError {
             code: Error::ExpectedComma,
-            position: Position { line: 3, col: 9 }
+            span: Span {
+                start: Position { line: 3, col: 9 },
+                end: Position { line: 3, col: 9 }
+            }
         }
     );
 
@@ -33,7 +39,10 @@
         ron::from_str::<Vec<i32>>(list_string).unwrap_err(),
         SpannedError {
             code: Error::ExpectedComma,
-            position: Position { line: 4, col: 9 }
+            span: Span {
+                start: Position { line: 4, col: 9 },
+                end: Position { line: 4, col: 9 }
+            }
         }
     );
 
@@ -46,7 +55,10 @@
         ron::from_str::<Test>(struct_string).unwrap_err(),
         SpannedError {
             code: Error::ExpectedComma,
-            position: Position { line: 3, col: 9 }
+            span: Span {
+                start: Position { line: 3, col: 9 },
+                end: Position { line: 3, col: 9 }
+            }
         }
     );
 
@@ -59,7 +71,10 @@
         ron::from_str::<std::collections::HashMap<String, i32>>(map_string).unwrap_err(),
         SpannedError {
             code: Error::ExpectedComma,
-            position: Position { line: 3, col: 9 }
+            span: Span {
+                start: Position { line: 3, col: 9 },
+                end: Position { line: 3, col: 9 }
+            }
         }
     );
 
@@ -72,9 +87,25 @@
         ron::from_str::<u8>(extensions_string).unwrap_err(),
         SpannedError {
             code: Error::ExpectedComma,
-            position: Position { line: 3, col: 9 }
+            span: Span {
+                start: Position { line: 2, col: 50 },
+                end: Position { line: 3, col: 9 }
+            }
         }
     );
+
+    #[cfg(feature = "internal-span-substring-test")]
+    check_error_span_inclusive::<u8>(
+        extensions_string,
+        Err(SpannedError {
+            code: Error::ExpectedComma,
+            span: Span {
+                start: Position { line: 2, col: 50 },
+                end: Position { line: 3, col: 9 },
+            },
+        }),
+        "\n        u",
+    );
 }
 
 #[test]